crates/autopilot/src/run_loop.rs

```diff
@@ -664,9 +701,10 @@ impl RunLoop {
         let deadline = current.saturating_add(max_blocks_wait);
         tracing::debug!(%current, %deadline, %auction_id, "waiting for tag");
         loop {
-            if self.eth.current_block().borrow().number > deadline {
-                break;
-            }
+            let block = ethrpc::block_stream::next_block(self.eth.current_block()).await;
+            // Run maintenance to ensure the system processed the last available block so
+            // it's possible to find the tx in the DB in the next line.
+            self.run_maintenance(&block).await;
```

Comment:
So now we are updating the solvable orders cache twice? And the second time, we execute it regardless of the `RunLoopMode`. Not sure I got the intention correctly.

Reply:
> This was not intended but I think doesn't really hurt either. Because now it can happen that we frontload the auction building step and don't have anything to do at the start of the next runloop.

But doesn't that mean that even with `RunLoopMode::Unsynchronized`, we now have that heavy SQL query right on the critical path?

Reply:
Yeah, you are probably right. 👍 
I decoupled the main system update from the auction building in [38c5000](https://github.com/cowprotocol/services/pull/2906/commits/38c500013b5c2fb4202e82053a6b704622434471).
Overall that leaves the new `Maintenance` struct in an awkward place but I would like delay cleaning that up (i.e. most likely removing the `SolvableOrdersCache` until it's fast enough to run without thinking about it) until we completely switch over to the new run mode.

Reply:
I agree with @squadgazzz that we should probably just call the `settlement_indexer` maintenance here. I agree that exposing a lot of public methods on maintenance is not ideal, but I think it's worth the tradeoff here (running a potentially 20s long solvable orders update on the critical path).

Otherwise, this also messes with the cutoff time to "miss a block in case too much time passed" logic in `next_auction` as a lot of time may have passed since we already updated all caches (and don't have to run maintenance anymore). Let's just do what we need here and have the main update in one place.